tensorrt_llm/models/llama/model.py

```diff
@@ -65,11 +65,16 @@ def __init__(self, config: LLaMAConfig, layer_idx: int):
             tp_rank=config.mapping.tp_rank,
             quant_mode=config.quant_mode)
 
-        mlp_hidden_size = config.hidden_size * 4 if config.intermediate_size is None else config.intermediate_size
+        if int(self.local_layer_idx) % 2 == 0:
```

Comment:
We can check layer_idx in moe_layers instead of assuming that moe_layers is perfectly interleaved.

Reply:
Good call, updated in the new commit for this place and all related.